path-walk.c

```diff
@@ -366,17 +380,10 @@ int walk_objects_by_path(struct path_walk_info *info)
 			if (t->object.flags & SEEN)
 				continue;
 			t->object.flags |= SEEN;
-
-			if (!oidset_insert(&root_tree_set, oid))
-				oid_array_append(&root_tree_list->oids, oid);
+			oid_array_append(&root_tree_list->oids, oid);
```

Comment:
Ah, we do not even need the `root_tree_set` anymore because the `SEEN` flag _already_ prevents duplicate entries. Nice!